### PR TITLE
[inductor] Use torch.cuda.clock_rate instead of triton.testing.nvsmi

### PR DESCRIPTION
**Pull request by bertmaher**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118662

`triton.testing.nvsmi` invokes `nvidia-smi` as a subprocess, and Meta
prod usually doesn't make nvidia-smi available.  Might as well just use
something that's native to torch.

Differential Revision: [D53235814](https://our.internmc.facebook.com/intern/diff/D53235814/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/utils.py [hunk 1]
```diff
@@ -1144,6 +1144,13 @@ def blue_text(msg):
     return _color_text(msg, "blue")
 
 
+def _max_clock_rate(msg):
+    import pynvml
```
Comment (jansel) [inline torch/_inductor/utils.py:1148]: Is this a new dependency?  Will it be installed for all OSS users?

**Review by jansel — CHANGES_REQUESTED** [review]
Fix lints

File: torch/_inductor/utils.py (hunk 1 above)
Comment (albanD) [reply torch/_inductor/utils.py:1148]: It is an optional dependency today that is only necessary for some specific monitoring tools (like temperature or process running on the GPU) https://github.com/pytorch/pytorch/blob/1128cf96f078a76f27b038d7adbc2b72fe8927e2/torch/cuda/__init__.py#L53C1-L53C12
We should not assume it is there indeed.

Comment (bertmaher) [reply torch/_inductor/utils.py:1148]: Ah, this is unfortunate - it's used by torch.cuda.clock_rate() and several other helps, so I assumed it was fair game, but it turns out it's not, and there's kind of an annoying history here.  Sigh.  Gonna just live with this failing internally.

**Comment by bertmaher** [comment]
Reviving this - it seems to fix an internal SEV.  I suspect what's going on is that when the call to nvsmi fails, the result of get_device_tflops doesn't get cached, and we end up making many calls to nvidia-smi, which in aggregate slow the process down unacceptably.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**